django/core/handlers/asgi.py

```diff
@@ -239,7 +268,7 @@ async def send_response(self, response, send):
         if response.streaming:
             # Access `__iter__` and not `streaming_content` directly in case
             # it has been overridden in a subclass.
-            for part in response:
+            async for part in _sync_to_async_iterator(response):
```

Comment:
Curious - what was your thought process behind writing an iterator converter rather than just wrapping this whole set of code inside `if` inside sync_to_async?

Reply:
To be honest, my original thought process happened when I was trying to get this working without any code changes to Django itself, so I was trying to make the smallest change to existing code as possible. Not exactly a relevant frame of mind anymore. I also thought it would be simpler, but it doesn't turn out that way because the usual tools like `sync_to_async` don't handle async generators.

In my testing of other solutions it seems that an `async_to_sync` call inside `sync_to_async` is a lot slower. I just tried your suggestion like this:

```python
            def process_sync_response():
                # Access `__iter__` and not `streaming_content` directly in
                # case it has been overridden in a subclass.
                for part in response:
                    for chunk, _ in self.chunk_bytes(part):
                        async_to_sync(send)({
                            'type': 'http.response.body',
                            'body': chunk,
                            # Ignore "more" as there may be more parts; instead,
                            # use an empty final closing message with False.
                            'more_body': True,
                        })
            await sync_to_async(process_sync_response)()
```

Using the same profiling as before, it runs in `16.2 msec per loop`, compared to `3.64 msec per loop` in the main branch. I don't have a good concept of how much those numbers matter, so no arguments if reducing the code complexity is worth the performance hit.

Reply:
Your solution honestly isn't _that_ much more complex, so I'm not saying we shouldn't do it, I was just curious how you ended up here! I think the resulting patch is pretty nice - I will need to take more time to properly review it but I like it on first glance.